### DISCUSSION TITLE
3-node cluster, the image queue mode is all, and the cluster is unavailable after partition

### DISCUSSION THREAD

**Discussion by wangdonghello** [rabbitmq/rabbitmq-server#4910] [Other]
3-node cluster,has all queues metadata per mode because the mirror queue mode is all(ha-mode = all),but after occure partition,the partition cluster don not work,why? The every mode has all queues metadata，it should be working normal

**Comment by michaelklishin** [comment]
Thank you for your time.

Team RabbitMQ uses GitHub issues for specific actionable items engineers can work on. This assumes that we have a certain amount of information to work with.

Getting all the details necessary to reproduce an issue, make a conclusion or even form a hypothesis about what's happening can take a fair amount of time. Our team is multiple orders of magnitude smaller than the RabbitMQ community. Please help others help you by providing a way to reproduce the behavior you're
observing and sharing as much relevant information as possible [on the list](https://groups.google.com/forum/#!forum/rabbitmq-users):

 * Server, client library and plugin (if applicable) versions used
 * Server logs
 * A code example or terminal transcript that can be used to reproduce
 * **Full** exception stack traces (not a single line message)
 * `rabbitmqctl status` (and, if possible, `rabbitmqctl environment` output)
 * Other relevant things about the environment and workload, e.g. OS/systemd logs, [a traffic capture](https://www.rabbitmq.com/amqp-wireshark.html), deployment tool logs, etc

Feel free to edit out hostnames and other potentially sensitive information.

When/if we have a complete enough understanding of what's going on, a recommendation will be provided or a new issues with more context will be filed.

Thank you.

**Comment by michaelklishin** [comment]
We cannot suggest anything with the amount of information provided. We don't have any information on what RabbitMQ version is used, on what the effective classic queue policies are, what's in the logs, how many nodes are disconnected, and so on.

What we can suggest with certainty is that you should **avoid using classic mirrored queues** and
go with [quorum queues](https://rabbitmq.com/quorum-queues.html) or [streams](https://rabbitmq.com/stream.html) instead. They use a predictable recovery algorithm,
are safer, more efficient at the same time (!), have lower memory footprint in 3.10, and are here to stay for the long term. Classic queue mirroring is **going to be removed** in RabbitMQ 4.0.
Mirroring documentation guide recommends quorum queues and says as much about their deprecation status.

There will be no more work put into classic queue mirroring.

**Comment by michaelklishin** [comment]
> The every mode has all queues metadata，it should be working normal

Having queue metadata is not enough. [Having a promotable mirror](https://rabbitmq.com/ha.html#unsynchronised-mirrors) is a very important factor for classic mirrored queues.

Quorum queues need a majority of nodes to be online and their algorithm more or less guarantees that an electable in that case (what Raft guarantees is the consistency of the log but we can oversimplify for the sake of this example).
replica will be available

**Comment by wangdonghello** [comment]
Thank you very much. We RabbitMQ version is 3.8.1。
Now, we find that the network packet loss occured network partition has serious problem，such as the queues become abnormal！

**Reply by michaelklishin** [reply]
You are over 30 release behind even in the 3.8 series.
